cli/azd/extensions/azure.foundry.ai.agents/internal/cmd/init.go

```diff
@@ -735,15 +738,40 @@ func (a *InitAction) downloadAgentYaml(
 
 func (a *InitAction) addToProject(ctx context.Context, targetDir string, agentManifest *agent_yaml.AgentManifest) error {
 	var host string
-	switch agentManifest.Agent.Kind {
+
+	// Convert the template to bytes
+	templateBytes, err := json.Marshal(agentManifest.Template)
+	if err != nil {
+		return fmt.Errorf("failed to marshal agent template to JSON: %w", err)
+	}
+
+	// Convert the bytes to a dictionary
+	var templateDict map[string]interface{}
+	if err := json.Unmarshal(templateBytes, &templateDict); err != nil {
+		return fmt.Errorf("failed to unmarshal agent template from JSON: %w", err)
+	}
+
+	// Convert the dictionary to bytes
+	dictJsonBytes, err := json.Marshal(templateDict)
+	if err != nil {
+		return fmt.Errorf("failed to marshal templateDict to JSON: %w", err)
+	}
+
+	// Convert the bytes to an Agent Definition
+	var agentDef agent_yaml.AgentDefinition
+	if err := json.Unmarshal(dictJsonBytes, &agentDef); err != nil {
+		return fmt.Errorf("failed to unmarshal JSON to AgentDefinition: %w", err)
+	}
+
+	switch agentDef.Kind {
 	case "container":
 		host = "containerapp"
 	default:
 		host = "foundry.agent"
```

Comment:
```suggestion
		host = "foundry.containeragent"
```

I think we renamed this recently so it should match what's in `listen.go`, `extension.yaml`, and `registry.json`